cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -309,15 +309,15 @@ func (c *Client) GetClusterVersions() (map[string]string, error) {
 
 // ClusterAvailable returns true if the cluster status indicates the cluster is available.
 func (c *Client) ClusterAvailable() (bool, error) {
-	_, err := c.GetClusterStatus()
+	_, err := c.GetClusterStatus(true)
 	if err != nil {
 		return false, err
 	}
 	return true, nil
 }
 
 // GetClusterStatus returns nil for status Up or error for status Down
-func (c *Client) GetClusterStatus() (map[string]*clientv3.StatusResponse, error) {
+func (c *Client) GetClusterStatus(noResponseMap bool) (map[string]*clientv3.StatusResponse, error) {
```

Comment:
If the previous comments are ok, this should change a little bit.
Probably we need a public method and a private method to accommodate all the need, but I leave to you to figure out the best way to implement this